tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -133,6 +148,104 @@ std::string ImportFunction(const std::string &functiondef_proto,
                                  status);
 }
 
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, TF_Status* status) {
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+  return ImportGraphDefImpl(proto, pass_pipeline, show_debug_info, debug_info,
+                            specs, status);
+}
+
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, absl::string_view input_names,
+                           absl::string_view input_data_types,
+                           absl::string_view input_data_shapes,
+                           absl::string_view output_names, TF_Status* status) {
+  std::vector<string> node_names = absl::StrSplit(input_names, ',');
+  std::vector<string> node_dtypes = absl::StrSplit(input_data_types, ',');
+  std::vector<string> node_shapes_str = absl::StrSplit(input_data_shapes, ':');
+  std::vector<std::vector<int>> node_shapes;
+
+  std::vector<int> dims;
+  for (const string& shape_str : node_shapes_str) {
+    dims.clear();
+    if(shape_str.empty()) continue;
+    for (const auto& dim_str : absl::StrSplit(shape_str, ',')) {
+      int size;
+      if (absl::SimpleAtoi(dim_str, &size)) {
+        dims.push_back(size);
+      } else {
+        auto s = tensorflow::errors::InvalidArgument("Invalid Shape Specified.",
+                                                     dim_str);
+        Set_TF_Status_from_Status(status, s);
+        return "// error";
+      }
+    }
+    node_shapes.push_back(dims);
+  }
+  std::vector<string> output_nodes = absl::StrSplit(output_names, ',');
```

Comment:
Looks like if `output_names` is empty (e.g. because `output_names=[]` was passed to `import_graphdef`) this results in an error:
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: Output  was not found in graph
```

It may be enough to do
```
  if (!output_names.empty()) {
    std::vector<string> output_nodes = absl::StrSplit(output_names, ',');
  }
```